pandas/core/groupby/generic.py

```diff
@@ -1710,3 +1714,69 @@ def _normalize_keyword_aggregation(kwargs):
         order.append((column,
                       com.get_callable_name(aggfunc) or aggfunc))
     return aggspec, columns, order
+
+
+def _make_lambda(func, i):
+    def f(*args, **kwargs):
+        return func(*args, **kwargs)
+    f.__name__ = "<lambda_{}>".format(i)
+    return f
+
+
+def _managle_lambda_list(aggfuncs):
+    i = 0
+    aggfuncs2 = []
+    for aggfunc in aggfuncs:
+        if com.get_callable_name(aggfunc) == "<lambda>":
+            if i > 0:
+                aggfunc = _make_lambda(aggfunc, i)
+            i += 1
+        aggfuncs2.append(aggfunc)
+
+    return aggfuncs2
+
+
+def _maybe_mangle_lambdas(agg_spec):
+    """
+    Make new lambdas with unique names.
+
+    Parameters
+    ----------
+    agg_spec : Any
+        An argument to NDFrameGroupBy.agg.
+        Non-dict-like `agg_spec` are pass through as is.
```

Comment:
we should try to type this (doesn't have to be here) to solidfiy what we actually accept to .agg; can you create an issue

Reply:
I think we have that one meta-issue for typing, and groupby should be on it.

Reply:
Not sure of the issue mentioned but new additions to code base would ideally included types if not too much extra effort (most of the current issues are regarding existing code)

Reply:
This one seems hard to get right (though I may be wrong).